service/controller/resource/statuscondition/create.go

```diff
@@ -220,7 +220,7 @@ func (r *Resource) computeCreateClusterStatusConditions(ctx context.Context, cr
 		isUpdating := status.HasUpdatingCondition()
 		notUpdated := !status.HasUpdatedCondition()
 		sameMasterCount := readyMasterReplicas != 0 && readyMasterReplicas == desiredMasterReplicas
-		sameWorkerCount := readyWorkerReplicas != 0 && readyMasterReplicas == desiredWorkerReplicas
+		sameWorkerCount := readyWorkerReplicas != 0 && readyWorkerReplicas == desiredWorkerReplicas
```

Comment:
Is this proven to be good enough or fixing the situation? We should have unit tests at best for this.

Reply:
I added the unittest for one case (set the updated condition), I know it's not the prettiest test and we can definitely improve it also add more tests for all conditions, but it ensures we set condition "Updated".